nipype/pipeline/engine/workflows.py

```diff
@@ -425,6 +425,9 @@ def write_graph(self, dotfilename='graph.dot', graph2use='hierarchical',
                 base_dir = os.getcwd()
         base_dir = make_output_dir(base_dir)
         if graph2use in ['hierarchical', 'colored']:
+            if self.name[:1].isdigit(): # these graphs break if int
+                raise ValueError('{} graph failed, workflow name cannot begin '
+                                 'with a number'.format(graph2use))
```

Comment:
@satra the error only occurs when the workflow graph is generated if graph2use is specified as either hierarchical/colored - the other options generate graphs just fine. I'm fine with moving the check to the `Workflow` initialization if we want to go in that direction - was trying to affect the smallest range of legacy workflows